src/main/java/com/squareup/javapoet/CodeBlock.java

```diff
@@ -102,6 +104,53 @@ public static CodeBlock of(String format, Object... args) {
     return new Builder().add(format, args).build();
   }
 
+  /**
+   * Joins {@code codeBlocks} into a single {@link CodeBlock}, each separated by {@code separator}.
+   * For example, joining {@code String s}, {@code Object o} and {@code int i} using {@code ", "}
+   * would produce {@code String s, Object o, int i}.
+   */
+  public static CodeBlock join(Iterable<CodeBlock> codeBlocks, String separator) {
+    Collection<CodeBlock> collection;
+    if (codeBlocks instanceof Collection) {
+      collection = (Collection<CodeBlock>) codeBlocks;
+    } else {
+      collection = new ArrayList<>();
+      codeBlocks.forEach(collection::add);
```

Comment:
Use `StreamSupport.stream(codeBlocks.spliterator(), false)` to create a stream from the iterable instead of making a copy?

Reply:
Good call

Reply:
I'm not sure why, but Github isn't picking up my latest commit which added this...

Reply:
That is weird. It's definitely on the branch that created this PR...

Reply:
I squashed them all into one commit and it seems to be getting it all now.